test/Microsoft.ML.Tests/MemoryCollectionTests.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.TestFramework;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Transforms;
+using System;
+using System.Collections.Generic;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.EntryPoints.Tests
+{
+    public class MemoryCollectionTests : BaseTestClass
+    {
+        public MemoryCollectionTests(ITestOutputHelper output)
+            : base(output)
+        {
+
+        }
+
+        [Fact]
+        public void ConstructorDoesntThrow()
+        {
+            Assert.NotNull(new MemoryCollection<Input>(new List<Input>()));
+            Assert.NotNull(new MemoryCollection<Input>(new List<Input>() { new Input { Number1 = 1, String1 = "1" } }));
+            Assert.NotNull(new MemoryCollection<Input>(new Input[0]));
+            Assert.NotNull(new MemoryCollection<Input>(new Input[1] { new Input { Number1 = 1, String1 = "1" } }));
+            Assert.NotNull(new MemoryCollection<Input>(null));
+        }
+
+        [Fact]
+        public void CanSuccessfullyApplyATransform()
+        {
+            var collection = new MemoryCollection<Input>(new List<Input>() { new Input { Number1 = 1, String1 = "1" } });
```

Comment:
>MemoryCollection<Input> [](start = 33, length = 23)

Just an observation... if we had instead structured this as a static utility method somewhere, then we could avoid having the double-specification of the `Input` class, as it would have been inferred by the compiler.

So: if we had a input type `MyAwesomeInputType`, then instead of `MemoryCollection<MyAwesomeInputType>(new MyAwesomeInputType[] {...})`, we would have `MemoryCollection.Create(new MyAwesomeInputType[] {...})` since the compiler could have done the work of inferring the type and whatnot.